src/panels/config/automation/trigger/types/ha-automation-trigger-calendar.ts

```diff
@@ -39,11 +39,13 @@ export class HaCalendarTrigger extends LitElement implements TriggerElement {
         ],
       ],
     },
+    { name: "offset", selector: { text: {} } },
```

Comment:
Should use the number selector with mode box or update to text type number.

https://github.com/home-assistant/frontend/blob/6fd4cda53488c570504f78a578ef3e7bd6baa6b8/src/data/selector.ts#L152

https://github.com/home-assistant/frontend/blob/6fd4cda53488c570504f78a578ef3e7bd6baa6b8/src/data/selector.ts#L181

Reply:
None of these are quite like the format that we use, or that Sun uses, like "-00:15:00" for 15 minutes, unless i'm misunderstanding how to use these. They just looked like plain numbers.  Should we add a string type that is specific for offsets?

Reply:
Oh i see these types are just built in html types...

Reply:
The duration selector actually makes more sense here I think, then add another input field for before/after toggle

Reply:
OK, i tried to do this as simple as possible, but it was a little harder than I was expecting. It doesn't look amazing since the selectors are each in their own row. However, not using ha-form was really complex, so I stuck with this, and am OK with it, but very open to other suggestions.